_posts/2019-03-09-rest-response-size.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title:  "Myth: REST Responses Sizes are massive!"
+date:   2019-03-04 19:08:05 +0000
+categories: jekyll update
+---
+
+One of the biggest complaints people have about REST is that the response size is too big. In reality though, that is not a REST problem, that is a problem with the developers maintaining that endpoint. Let's look at some reasons why and how we can mitigate it. 
+
+## Sending too much data
+
+Thoughtfully developing an API endpoint requires some thought into what data is or is not being sent over the wire. While it takes little effort to add a new field to your response object, it does take some thought into _if_ the field should be added in the first place. One of the best ways to mitigate this is by embracing Hypermedia with your APIS. Lets use an example of a users response object before we add Hypermedia. 
+
+```json
+{
+  "data": [
+    {
+      "type": "User",
+      "id": 1,
+      "attributes": {
+        "userId": "11122",
+        "name": "Crashy McCrashface",
+        "position": "Lead Bike Engineer",
+        "yearsAtPosition": 2,
+        "previousPosition": "Bike Engineer",
+        "managerId": "10109"
+      }
+    },
+    {
+      "type": "User",
+      "id": 2,
+      "attributes": {
+        "userId": "11123",
+        "name": "Turtle McTurtleface",
+        "position": "Lead Swamp Designer",
+        "yearsAtPosition": 5,
+        "previousPosition": "Junior Swamp Designer",
+        "managerId": "10002"
+      }
+    }
+  ];
+}
+```
+
+Looking at the data above, there is so much included that doesnt need to be in a ```HTTP/2 GET /users``` request. Lets simplify the whole thing. 
+
+```json
+{
+  "data": [
+    {
+      "type": "User",
+      "id": 1,
+      "attributes": {
+        "userId": "11122",
+        "name": "Crashy McCrashface"
+      },
+      "relationships": {
+        "user": {
+          "links": {
+            "self": "https://apisyouwonthate.com/api/users/11122/relationships/user",
```

Comment:
I wonder if this is just begging for confusion. JSON:API does not always appear to be the most concise or obvious thing in the world, so trying to use this as an example here might just complicate the explanation. I'm not sure there. 

Are we trying to move the manager info out? 

Perhaps a better example would be invoices. People often mix invoices, with payments and payment attempts, and status of those payments as it traces through the bank, which is confusing as fuck.

Reply:
Yea, I agree.